File: sdks/python/resolver_test.py
from cyanprintsdk.domain.resolver.input import ResolverInput
from cyanprintsdk.domain.resolver.output import ResolverOutput
from cyanprintsdk.main import start_resolver_with_fn


async def resolver(i: ResolverInput) -> ResolverOutput:
    print("Config:", i.config)
    print("Files:", len(i.files))

    # Simple merge: just return the first file's content as-is (placeholder implementation)
    if not i.files:
        return ResolverOutput(path="", content="")

    first_file = i.files[0]
    return ResolverOutput(path=first_file.path, content=first_file.content)


start_resolver_with_fn(resolver)

File: sdks/node/resolver_test.ts
import { StartResolverWithLambda, type ResolverInput, type ResolverOutput } from './src/main.js';

StartResolverWithLambda(async (i: ResolverInput): Promise<ResolverOutput> => {
  console.log('Config:', i.config);
  console.log('Files:', i.files.length);

  // Simple merge: just return the first file's content as-is (placeholder implementation)
  const firstFile = i.files[0];
  if (!firstFile) {
    return { path: '', content: '' };
  }

  return { path: firstFile.path, content: firstFile.content };
});
